wire-compiler/src/main/java/com/squareup/wire/java/TypeWriter.java

```diff
@@ -340,12 +339,7 @@ private MethodSpec messageEquals(WireMessage type) {
     result.addStatement("if (other == this) return true");
     result.addStatement("if (!(other instanceof $T)) return false", javaType);
 
-    // TODO(jwilson): extensions!
-    //if (compiler.hasExtensions(messageType)) {
-    //  writer.emitStatement("if (!extensionsEqual(o)) return false");
-    //}
-
-    if (fields.size() == 1) {
+    if (fields.size() == 1 && type.extensions().isEmpty()) {
```

Comment:
Is this special case even worth it anymore? The byte code is equivalent, it just saves one line of source.